dashboard/app/helpers/admin_reports_helper.rb

```diff
@@ -11,4 +11,36 @@ def get_gender_percentage(this_gender_count, other_gender_count)
   def get_percentage_from_ratio(ratio)
     return number_with_precision(100.0 * ratio, precision: 2).to_s + '%'
   end
+
+  # From a hash of hashes giving counts, get_cumulatives constructs a hash of hashes giving
+  # cumulative counts, e.g., transforms stats[key][subkey] = count to cumulatives[key][subkey] =
+  # cumulative_count, where cumulative_count := SUM(stats[key][i] : i <= subkey).
+  # The max_counts parameter is used to determine the space of subkeys, e.g.,
+  # cumulatives[key][subkey] will exist exactly when subkey <= max_counts[key].
+  def get_cumulatives(max_counts, stats)
```

Comment:
The parameters are being piped in to this method from data stored in the Pegasus Properties table. There, for readability and clarity (being in the DB), it seems useful to make the data more self-documenting as a hash rather than array. If you feel otherwise, feel free to push back.